index.bs

```diff
@@ -5483,6 +5504,379 @@ At this time, one [=credential property=] is defined: the [=resident key credent
 :: None.
 
 
+# User Agent Automation # {#sctn-automation}
+
+For the purposes of user agent automation and [=web application=] testing, this document defines a number of [[WebDriver]] [=extension commands=].
+
+## <dfn>Virtual Authenticators</dfn> ## {#sctn-automation-virtual-authenticators}
+
+These WebDriver [=extension commands=] create and interact with [=Virtual Authenticators=]: software implementations of the
+<a>Authenticator Model</a>. <a>Virtual Authenticators</a> are stored in a <dfn>Virtual Authenticator Database</dfn>.
+Each stored [=virtual authenticator=] has the following properties:
+
+: <dfn>authenticatorId</dfn>
+:: An non-null string made using up to 48 characters from the `unreserved` production defined in Appendix A of [[RFC3986]]
+    that uniquely identifies the [=Virtual Authenticator=].
+
+    Note: [=generating a uuid|Generating this string as a UUID=] is recommended.
```

Comment:
I'm having 2nd thoughts about this unqualified recommendation of gen'g a UUID for authnrID -- it is seeming like overkill to me now.  In an isolated environment one could just number the authnrs 1, 2, 3, etc, yes?  plus different "sessions" are disambiguated by `session id`, so they won't have authnrID collisions.  what do you think?

Reply:
Sounds reasonable. Removed this note and reference.